apps/frontend/src/pages/project/project.tsx

```diff
@@ -1,54 +1,85 @@
 import {
 	Breadcrumbs,
 	Button,
+	Modal,
 	PageLayout,
 } from "~/libs/components/components.js";
 import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
 import {
 	useAppDispatch,
 	useAppSelector,
+	useCallback,
 	useEffect,
 	useModal,
 	useParams,
 } from "~/libs/hooks/hooks.js";
-import { actions as projectActions } from "~/modules/projects/projects.js";
+import {
+	actions as projectActions,
+	type ProjectPatchRequestDto,
+} from "~/modules/projects/projects.js";
 import { NotFound } from "~/pages/not-found/not-found.jsx";
+import { ProjectUpdateForm } from "~/pages/projects/libs/components/components.js";
 
-import { SetupAnalyticsModal } from "./libs/components/components.js";
+import {
+	ProjectMenu,
+	SetupAnalyticsModal,
+} from "./libs/components/components.js";
 import styles from "./styles.module.css";
 
 const Project = (): JSX.Element => {
 	const dispatch = useAppDispatch();
 	const { id: projectId } = useParams<{ id: string }>();
 
-	const { project, projectStatus } = useAppSelector(({ projects }) => projects);
+	const { project, projectPatchStatus, projectStatus } = useAppSelector(
+		({ projects }) => projects,
+	);
 
 	const {
 		isOpened: isSetupAnalyticsModalOpened,
 		onClose: onSetupAnalyticsModalClose,
 		onOpen: onSetupAnalyticsModalOpen,
 	} = useModal();
 
+	const {
+		isOpened: isEditModalOpen,
+		onClose: handleEditModalClose,
+		onOpen: handleEditModalOpen,
+	} = useModal();
+
 	useEffect(() => {
 		if (projectId) {
 			void dispatch(projectActions.getById({ id: projectId }));
+
+			if (projectPatchStatus === DataStatus.FULFILLED) {
+				handleEditModalClose();
+			}
 		}
-	}, [dispatch, projectId]);
+	}, [dispatch, projectId, projectPatchStatus, handleEditModalClose]);
+
+	const handleEditProject = useCallback(() => {
+		handleEditModalOpen();
+	}, [handleEditModalOpen]);
+
+	const handleProjectEditSubmit = useCallback(
+		(payload: ProjectPatchRequestDto) => {
+			if (project) {
+				void dispatch(projectActions.patch({ id: project.id, payload }));
+			}
+		},
+		[dispatch, project],
+	);
 
 	const isLoading =
 		projectStatus === DataStatus.PENDING || projectStatus === DataStatus.IDLE;
-
 	const isRejected = projectStatus === DataStatus.REJECTED;
 
-	const hasProject = project !== null;
-
 	if (isRejected) {
 		return <NotFound />;
 	}
 
 	return (
 		<PageLayout isLoading={isLoading}>
-			{hasProject && (
+			{project && (
```

Comment:
Why was hasProject removed in favor of checking project directly? I think using hasProject improves readability and matches the structure of the app.

Reply:
Agree, @Fjortis have a look at this [comment](https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/171#discussion_r1742413987)

Reply:
I adjusted this, thanks for the reference